actions/setup-conftest/action.yaml

```diff
@@ -0,0 +1,51 @@
+name: Install conftest
+description: Install conftest
+
+inputs:
+  version:
+    description: |
+      Version of conftest to install.
+    default: 0.55.0
+
+runs:
+  using: composite
+
+  steps:
+    - name: Restore cache
+      id: restore
+      uses: actions/cache/restore@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+      with:
+        path: ${{ github.workspace }}/bin
+        key: conftest-linux-amd64-${{ inputs.version }}
+
+    - name: Fetch Github Release Asset
+      id: fetch_asset
+      if: steps.restore.outputs.cache-hit != 'true'
+      uses: dsaltares/fetch-gh-release-asset@a40c8b4a0471f9ab81bdf73a010f74cc51476ad4 # 1.1.1
+      with:
+        repo: "open-policy-agent/conftest"
+        version: "tags/v${{ inputs.version }}"
+        file: "conftest_${{ inputs.version }}_Linux_x86_64.tar.gz"
+        target: ${{ github.workspace }}/bin/conftest.tgz
+
+    - name: Unpack tarball
+      id: unpack
+      if: steps.fetch_asset.outcome == 'success'
+      shell: sh
+      run: |
+        tar -zxvf ${{ github.workspace }}/bin/conftest.tgz conftest
+        mv conftest ${{ github.workspace }}/bin
+        chmod +x ${{ github.workspace }}/bin/conftest
+
+    - name: Save to cache
+      id: save
+      if: steps.gunzip.outcome == 'success'
+      uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+      with:
+        path: ${{ github.workspace }}/bin
+        key: ${{ steps.restore.outputs.cache-primary-key }}
```

Comment:
AFAICS you could use `actions/cache` instead of the separate `restore` and `save` and get the right behaviour, would save a bit of code.